### PR TITLE
refactor: move shared messages to errors.h

### PR DESCRIPTION
**Pull request by justinmk**
Just a minor cleanup. If it's preferred in globals.h for some reason, we can close this.

~~Another idea is to introduce `messages.h` (plural), but that would be more noisy.~~

### CODE REVIEW

**Review by dundargoc — APPROVED** [review]

**Comment by zeertzjq** [comment]
Isn't it better to add an `errors.h` like Vim?

**Comment by justinmk** [comment]
> Isn't it better to add an `errors.h` like Vim? 

done

> (With the exception of `top_bot_msg`, `bot_top_msg` and `line_msg` which are still in `globals.h`)

I assume that wasn't intentional. So I put those in errors.h.

File: src/nvim/api/private/dispatch.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #include <stdint.h>
 
 #include "nvim/api/private/defs.h"  // IWYU pragma: keep
+#include "nvim/errors.h"  // NOLINT(build/include_defs)
```
Comment (justinmk) [inline src/nvim/api/private/dispatch.h:7]: Should it be errors_defs.h? This NOLINT seems to be ignored by clint. 

Without this include, linker fails:

```
ack-protector-strong  -lm  -lutil  .../neovim/.deps/usr/lib/libuv.a  -ldl && :
Undefined symbols for architecture x86_64:
  "_bot_top_msg", referenced from:
      _searchit in search.c.o
      _search_for_exact_line in search.c.o
      _spell_move_to in spell.c.o
  "_e_api_error", referenced from:
      _api_wrapper in funcs.c.o
  "_e_api_spawn_failed", referenced from:
      _f_rpcstart in funcs.c.o
  "_e_argreq", referenced from:
      _do_augroup in autocmd.c.o
      _do_write in ex_cmds.c.o
      _do_one_cmd in ex_docmd.c.o
...
```

Comment (zeertzjq) [reply src/nvim/api/private/dispatch.h:7]: Include in in main.c

**Pull request merged**